src/ex3/Lattice.java

```diff
@@ -0,0 +1,49 @@
+package ex3;
+
+import java.util.*;
+
+public class Lattice {
+	private List<LatticeVar> set;
+
+	public Lattice() {
+		set = new ArrayList<LatticeVar>();
+	}
+
+	public void add(String varID, initKind kind) {
+		LatticeVar var = new LatticeVar(varID, kind);
+		set.add(var);
+	}
+
+	public void assign(String varID) {
+		for (LatticeVar var : set) {
+			if (varID.equals(var.VarID)) {
+				var.isInit = initKind.tt;
+				break;
+			}
+		}
+	}
+
+	public void join(Lattice l) {
+		for (int i = 0; i < set.size(); i++) {
+			set.get(i).isInit = LatticeVar.varJoin(set.get(i), l.set.get(i));
+		}
+	}
+
+	public boolean isInit(String varID) {
+		for (LatticeVar var : set) {
+			if (varID.equals(var.VarID)) {
+				if (var.isInit == initKind.tt)
+					return true;
+				else
+					return false;
+			}
+		}
+		return true;

```

Comment:
Shouldn't this be false?

Reply:
at this specific line "41" I reach the end of for loop so I didn't find the variable in my lattice struct, that means this variable is either a field or a formal parameter (as they are already initialized and I assume it couldn't be a variable that wasn't declared before)